File: Algorithm/baekjoon/10845_큐.py
# baekjoon source = "https://www.acmicpc.net/problem/10845"
import sys

def my_push(x):
    q.append(x)

def my_pop():
    if q:
        print(q.pop(0))
    else:
        print(-1)

def my_size():
    print(len(q))

def my_empty():
    if q:
        print(0)
    else:
        print(1)

def my_front():
    if q:
        print(q[0])
    else:
        print(-1)

def my_back():
    if q:
        print(q[len(q)-1])
    else:
        print(-1)

N = int(sys.stdin.readline())
q = []

for i in range(N):
    inp = sys.stdin.readline().split()
    if len(inp) == 1:
        if inp[0] == 'pop':
            my_pop()
        elif inp[0] == 'size':
            my_size()
        elif inp[0] == 'empty':
            my_empty()
        elif inp[0] == 'front':
            my_front()
        else:
            my_back()
    else:
        my_push(int(inp[1]))

File: Java/baekjoon/화목한스터디/src/큐/_10845_큐.java
//baekjoon source = "https://www.acmicpc.net/problem/10845"
package ť;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.StringTokenizer;

public class _10845_ť {

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
		int[] q = new int[10000];
		int top = -1, rear = -1;
		int n = Integer.parseInt(br.readLine());
		for (int i = 0; i < n; i++) {

			StringTokenizer st = new StringTokenizer(br.readLine());
			switch (st.nextToken()) {
			case "push":
				q[++rear] = Integer.parseInt(st.nextToken());
				break;
			case "pop":
				bw.write((top == rear ? -1 : q[++top]) + "\n");
				break;
			case "size":
				bw.write(rear - top + "\n");
				break;
			case "empty":
				bw.write((top == rear ? 1 : 0) + "\n");
				break;
			case "front":
				bw.write((top == rear ? -1 : q[top + 1]) + "\n");
				break;
			case "back":
				bw.write((top == rear ? -1 : q[rear]) + "\n");
			}
		}
		bw.flush();
	}
}
